cpp/src/join/hash_join.cuh

```diff
@@ -0,0 +1,331 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/table/table_view.hpp>
+#include <cudf/table/table.hpp>
+#include <cudf/table/table_device_view.cuh>
+#include <hash/concurrent_unordered_multimap.cuh>
+#include <cudf/scalar/scalar.hpp>
+#include <cudf/scalar/scalar_device_view.cuh>
+#include <cudf/detail/utilities/cuda.cuh>
+
+#include <join/join_common_utils.hpp>
+#include <join/join_kernels.cuh>
+#include <join/full_join.cuh>
+
+namespace cudf {
+
+namespace detail {
+
+using join_type = ::cudf::detail::join_type;
```

Comment:
These statements look weird to me. Isn't this a no-op?

Reply:
Removed